### DISCUSSION TITLE
base path api url different from host

### DISCUSSION THREAD

**Discussion by olick** [swagger-api/swagger-ui#8555] [Q&A]
Hello everyone, 
When I'm calling method form SwaggerUI I would like to override the base path of url use by swagger. 

This is how I'm loading SwaggerUI 

```
import SwaggerUI from 'swagger-ui';
//[...]
  SwaggerUI({
                url, // => https://mydomain.priv/auth/api/swagger/v1/swagger.json
                dom_id: '#swagger-ui',
                requestInterceptor: (req) => {
                    req.headers.Authorization = authorization;
                    return req;
                },
            });
```
So, I'm expecting Swagger to call 
https://mydomain.priv/auth/api/v1/controller/method, instead of
https://mydomain.priv/v1/controller/method
How can I tell swagger to use /auth/api/ ?

**Comment by cihatsolak** [comment]
Hi,

Why don't you make this request by specifying the base path? Do you have any hesitation? I could not fully understand.  Can you check this link?  https://swagger.io/docs/specification/2-0/api-host-and-base-path/

Also, does this method work for you?

```
SwaggerUI({
  url,
  dom_id: '#swagger-ui',
  requestInterceptor: (req) => {
    req.headers.Authorization = authorization;
    return req;
  },
  basePath: '/auth/api'
});
```
This issue has been closed. maybe you want to check it out.
https://github.com/scottie1984/swagger-ui-express/issues/189

**Reply by olick** [reply]
Hello, thanks for the answer.

I dont understand your question. 
_Why don't you make this request by specifying the base path?_
If you mean why I'm not calling the url directly is because my apis are located in a different url that my front-end

_Do you have any hesitation?_
What do you mean ? Hesitation to do what ? 

_I could not fully understand_
Maybe this will help you to understand : 

```
ngOnInit(): void {
        this.formSwaggerControl.valueChanges.subscribe((apiSwagger: SwaggerApiModel): void => {
            const currentToken = this._authenticationService.getToken();
            let url: string = null;

            switch (apiSwagger.enumNumber) {
                case ApiEnum.ApiA: {
                    url = 'https://mydomain.priv/auth/api-a/swagger/v1/swagger.json';
                    break;
                }
                case ApiEnum.ApiB: {
                    url = 'https://mydomain.priv/auth/api-b/swagger/v1/swagger.json';
                    break;
                }
                case ApiEnum.ApiC: {
                      url = 'https://mydomain.priv/auth/api-c/swagger/v1/swagger.json';
                    break;
                }
                default: {
                    break;
                }
            }

            const authorization = `Bearer ${currentToken}`;

            SwaggerUI({
                url,
                dom_id: '#swagger-ui',
                presets: [SwaggerUI.presets.apis],
                plugins: [SwaggerUI.plugins.DownloadUrl],
                requestInterceptor: (req) => {
                    req.headers.Authorization = authorization;
                    req.headers.Accept = 'application/json'; // Ajouter l'en-tête Accept
                    return req;
                },
               // basePath : //// whatever path I put in there is not changing anything
            });
        });
    }

```

Like I said, my apis are located to : 
https://mydomain.priv/auth/api-[a|b|c]/v1/controller/method
When I'm triggering the "try it out / execute" buttons of swagger-ui, it is calling : 
https://mydomain.priv/v1/controller/method

I have spotted the base path in source code, so thats why I was asking if I could precise the correct path to use to swagger ui 

_Can you check this link?_
Of course, thanks, I've already checked that link :) 

_This issue has been closed. maybe you want to check it out._ 
Thanks, I didnt see this link even after looked out. I have to stay in openapi 3.0 definition unfortunately... I'm still gonna try what they propose.
